src/app/settings/views/Configuration/Security/VaultSettings/VaultSettings.tsx

```diff
@@ -0,0 +1,144 @@
+
+  if (vaultEnabled) {
+    return (
+      <>
+        <p>
+          <Icon name="security-tick" />
+          <span className="u-nudge-right--small">Vault enabled</span>
+        </p>
+        <a href={docsUrls.aboutNativeTLS}>More about Vault integration</a>
+      </>
+    );
+  } else
+    return (
+      <>
+        {unconfiguredControllers.length >= 1 ? (
+          <>
+            {configuredControllers.length >= 1 ? (
+              <p>
+                <Icon name="security-warning" />
+                <span className="u-nudge-right--small">
+                  Incomplete Vault integration, configure{" "}
+                  {unconfiguredControllers.length} other{" "}
+                  {unconfiguredControllers.length > 1
+                    ? "controllers"
+                    : "controller"}{" "}
+                  with Vault to complete this operation.
+                </span>
+              </p>
+            ) : (
+              <p>
+                <Icon name="security" />
+                <span className="u-nudge-right--small">
+                  Integrate with Vault
+                </span>
+              </p>
+            )}
+            <div aria-label={Labels.SetupInstructions}>
+              <p>
+                1. Get the $wrapped_token and $role_id from Vault.{" "}
+                <a
+                  href="https://learn.hashicorp.com/tutorials/vault/approle-best-practices?in=vault/auth-methods#approle-response-wrapping"
+                  rel="noreferrer noopener"
+                  target="_blank"
+                >
+                  Find out more from Hashicorp Vault
+                </a>
+                .
+              </p>
+              <p>2. SSH into each region controller and configure Vault.</p>
+              <CodeSnippet
+                blocks={[
+                  {
+                    appearance: CodeSnippetBlockAppearance.LINUX_PROMPT,
+                    code: "sudo maas config-vault configure $url $approle_id $wrapped_token $secrets_path --secrets-mount $secret_mount",
+                  },
+                ]}
+              />
+              <p>
+                3. After Vault is configured on all region controllers, migrate
+                secrets on one of the region controllers.
+              </p>
+              <CodeSnippet
+                blocks={[
+                  {
+                    appearance: CodeSnippetBlockAppearance.LINUX_PROMPT,
+                    code: "sudo maas config-vault migrate",
+                  },
+                ]}
+              />
+            </div>
+          </>
+        ) : (
+          <>
+            <p>
+              <Icon name="security-warning" />
+              <span className="u-nudge-right--small">
+                Incomplete Vault integration, migrate secrets on one region
+                controller to complete setup.
+              </span>
+            </p>
+            <div aria-label={Labels.SecretMigrationInsctructions}>
```

Comment:
Ideally this should be a semantic element.

```suggestion
            <section aria-label={Labels.SecretMigrationInsctructions}>
```

Even better if you could you use the text that's displayed ("Incomplete Vault integration, migrate secrets on one region controller to complete setup.") and use an aria-labelledby  instead of a custom aria-label as I suggested for the other scenario above.

Reply:
Can do, committed now